plugin.cpp

```diff
@@ -81,16 +83,22 @@ Value CellularAutomata1D::nextState(const Node& leftNode, const Node& node, cons
     bool right = rightNode.attr(m_stateAttrId).toBool();
 
     bool r = false;
-    if (m_rule == 30) {
-        r = left ^ (center || right);
-    } else if (m_rule == 32) {
-        r = left && !center && right;
-    } else if (m_rule == 110) {
-        r = (!left && center) || (center ^ right);
-    } else if (m_rule == 250) {
-        r = left || right;
-    } else {
-        qFatal("invalid rule");
+    if (left && center && right) {
+        r = m_binrule.at(0) == "1";
+    } else if (left && center && !right) {
+        r = m_binrule.at(1) == "1";
+    } else if (left && !center && right) {
+        r = m_binrule.at(2) == "1";
+    } else if (left && !center && !right) {
+        r = m_binrule.at(3) == "1";
+    } else if (!left && center && right) {
+        r = m_binrule.at(4) == "1";
+    } else if (!left && center && !right) {
+        r = m_binrule.at(5) == "1";
+    } else if (!left && !center && right) {
+        r = m_binrule.at(6) == "1";
+    } else if (!left && !center && !right) {
+        r = m_binrule.at(7) == "1";
     }
```

Comment:
You could replace all those if-else statements by:
``` c++
return Value(m_binrule[left*4 + center*2 + right]);
```

Reply:
That's really clever and elegant! I wanted to use some sort of hash to access the cell state, but decided against it since I figured it would be an overkill (and an inefficient one at that).